### PR TITLE
Docs: no jinja nesting rule

### PR DESCRIPTION
**Pull request by oraNod**
Add MD documentation for the no-jinja-nesting rule.

### CODE REVIEW

File: src/ansiblelint/rules/no_log_password.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  hosts: localhost
+  tasks:
+    - name: Log user passwords
+      ansible.builtin.user:
+        name: johnd
+        comment: John Doe
+        uid: 1040
+        group: admin
+        password: password
+      no_log: False # <- Sets the no_log attribute to false.
```
Comment (oraNod) [inline src/ansiblelint/rules/no_log_password.md:20]: @ssbarnea Again I hope you don't read this while on PTO. But I wasn't able to trigger this rule, even using the tests. Am I missing something here? Thanks.

File: src/ansiblelint/rules/no_jinja_nesting.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+```yaml
+---
+# Uses the hostvars variable.
+{{ hostvars[inventory_hostname]['somevar_' ~ other_var] }}
+
+# Uses the ansible.builtin.vars lookup plugin.
+{{ lookup('vars', 'somevar_' ~ other_var) }}
+```
```
Comment (oraNod) [inline src/ansiblelint/rules/no_jinja_nesting.md:35]: @ssbarnea One other question I have in this PR is about contextualizing these examples into playbooks. I found this one a little difficult and worry any playbook example might be nonsense. I'm still fairly new to Ansible and Jinja. Do we have "advanced" examples anywhere?

Comment (ssbarnea) [reply src/ansiblelint/rules/no_jinja_nesting.md:35]: While looking at this rule I realised that I will probably remove this rule by encapsulating the feature it provides into `jinja` one.

I will keep this PR open until I have the proof that I can bring that functionality into the jinja rule, especially as that would make it much more trustable (avoiding the regex, which brings false positivies)

**Comment by ssbarnea** [comment]
Closing as #2436 removed this rule.

**Pull request closed without merging**